src/main/java/frc/robot/Constants.java

```diff
@@ -12,4 +12,45 @@
  * <p>It is advised to statically import this class (or one of its inner classes) wherever the
  * constants are needed, to reduce verbosity.
  */
-public final class Constants {}
+public final class Constants {
+    //TODO: decide whether to move ModuleConstants to SwerveModule class
+    public static final class ModuleConstants {
+        public static final double kWheelDiameterMeters = .1016; //in meters //4 inches
+        public static final double kDriveMotorGearRatio = 1/5.84628; //TODO: measure drive motor gear ratio
+        public static final double kSteerMotorGearRatio = 1/18.0; //TODO: measure steer motor gear ratio
+        public static final double kDriveEncoderRotToMeters = kDriveMotorGearRatio * Math.PI * kWheelDiameterMeters;
+        public static final double kSteerEncoderRot2Rad = kSteerMotorGearRatio *kDriveEncoderRotToMeters;
+        public static final double kDriveEncoderRpm2Mps = kDriveEncoderRotToMeters / 60; //rpm = rotations per minute //mps = meters per second
+        public static final double kSteerEncoderRPM2RadPerSec = kSteerEncoderRot2Rad / 60;
+        public static final double kCANCoderCounts = 4096.0; // CANCoders have a resolution of 4096 counts per revolution
+        public static final double kAbsoluteEncoderCountsPerMin2Rad = 2*Math.PI/kCANCoderCounts; 
```

Comment:
Also a good practice to have a trailing `.0` when the number is meant to be a double in math that involves doubles. What I mean is that it should be `60.0` and `2.0`.

Compilers are smart, so its not a big deal, but its a good habit to be very intentional about whether you are using integer constants or floating point constants.

Reply:
Alright, changed that and I'll try to be more intentional going forward.